portal/src/util/misc.ts

```diff
@@ -0,0 +1,17 @@
+export function isArrayEqualInOrder(arr1: unknown[], arr2: unknown[]): boolean {
+  if (arr1.length !== arr2.length) {
+    return false;
+  }
+  return arr1.every((elem, index) => elem === arr2[index]);
+}
+
+export function setFieldIfChanged(
```

Comment:
A better typing for this function.

```typescript
function setFieldIfChanged<K extends string>(record: Record<K, unknown>, key: K, initialValue: unknown, value: unknown): void
```

So we will not make typo mistake of the keys of the records.